### ISSUE TITLE
Plugin is broken with AGP 8.4+

### ISSUE THREAD

**Issue by dant3** [mannodermaus/android-junit5#352]
When trying to build a project using this plugin and AGP 8.4.0 or newer I'm getting the following error:

```
* What went wrong:
Some problems were found with the configuration of task ':libraries:android-coroutine-extensions:writeFiltersDebugAndroidTest' (type 'AndroidJUnit5WriteFilters').
  - Gradle detected a problem with the following location: '/Users/me/project/libraries/android-coroutine-extensions/build/generated/res/android-junit5/debugAndroidTest'.

    Reason: Task ':libraries:android-coroutine-extensions:generateDebugLintReportModel' uses this output of task ':libraries:android-coroutine-extensions:writeFiltersDebugAndroidTest' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

    Possible solutions:
      1. Declare task ':libraries:android-coroutine-extensions:writeFiltersDebugAndroidTest' as an input of ':libraries:android-coroutine-extensions:generateDebugLintReportModel'.
      2. Declare an explicit dependency on ':libraries:android-coroutine-extensions:writeFiltersDebugAndroidTest' from ':libraries:android-coroutine-extensions:generateDebugLintReportModel' using Task#dependsOn.
      3. Declare an explicit dependency on ':libraries:android-coroutine-extensions:writeFiltersDebugAndroidTest' from ':libraries:android-coroutine-extensions:generateDebugLintReportModel' using Task#mustRunAfter.
```

**Comment by mannodermaus** [comment]
Thanks for reporting. Having tried AGP 8.4 and 8.5, as well as plugin 1.10 and 1.11, I can't see this with any of these combinations in my projects, so I'll need to get some more info on your setup.

- What's your Gradle version?
- What Gradle task do you execute that leads to the error posted above?
- Do you use JUnit 5 for instrumentation tests?
  - If yes, do you have any test filters configured in your project (via the `junitPlatform` block)?

Thanks in advance.

**Comment by bddckr** [comment]
I think I've run into the same problem. It looks like the Android Lint plugin is at fault instead.
```kotlin
// Work around https://issuetracker.google.com/issues/337776938
tasks
    .named {
        it == "generateDebugAndroidTestLintModel" || it == "lintAnalyzeDebugAndroidTest"
    }
    .configureEach { dependsOn("writeFiltersDebugAndroidTest") }
```

**Comment by bddckr** [comment]
[AGP 8.6.0 and Android Studio Koala Feature Drop | 2024.1.2 are out.](https://androidstudio.googleblog.com/2024/08/android-studio-koala-feature-drop_29.html) That AGP version has a fix for this issue, and the workaround I posted is no longer necessary.

**Comment by mannodermaus** [comment]
Thank you for the thorough investigations, y'all! Not sure if I want to try and inject a fix for AGP 8.4 and 8.5 into the plugin itself, since I still can't seem to repro this myself. Good to know that 8.6 resolved it, though!
